packages/cli/src/middlewares/listQuery/dtos/workflow.filter.dto.ts

```diff
@@ -0,0 +1,40 @@
+import { isObjectLiteral } from '@/utils';
+import { IsOptional, IsString, IsBoolean, IsArray, validate } from 'class-validator';
+import { instanceToPlain } from 'class-transformer';
+
+export class WorkflowFilter {
+	@IsString()
+	@IsOptional()
+	name: string;
+
+	@IsBoolean()
+	@IsOptional()
+	active?: boolean;
+
+	@IsArray()
+	@IsString({ each: true })
+	@IsOptional()
+	tags?: string[];
+
+	static get filterableFields() {
+		return new Set(['name', 'active', 'tags']);
+	}
+
+	constructor(dto: unknown) {
+		if (!isObjectLiteral(dto)) throw new Error('Filter must be an object literal');
+
+		const onlyKnownFields = Object.fromEntries(
+			Object.entries(dto).filter(([key]) => WorkflowFilter.filterableFields.has(key)),
+		);
+
+		Object.assign(this, onlyKnownFields);
+	}
+
```

Comment:
could this be accomplished using `plainToClass`/`plainToInstance` ?

Reply:
Please see [`60b4990`](https://github.com/n8n-io/n8n/pull/6892/commits/60b49908f9723408ff3f2b9e1d1dc83b0cfc3c31).